opentelemetry-sdk/tests/metrics/exponential_histogram/test_exponential_bucket_histogram_aggregation.py

```diff
@@ -920,21 +925,68 @@ def test_collect_results_cumulative(self):
             collection_1.positive.bucket_counts,
             [
                 1,
-                *[0] * 15,
+                *[0] * 17,
                 1,
-                *[0] * 47,
+                *[0] * 36,
                 1,
-                *[0] * 40,
+                *[0] * 15,
+                2,
+                *[0] * 15,
                 1,
-                *[0] * 17,
+                *[0] * 15,
                 1,
-                *[0] * 36,
+                *[0] * 15,
+                1,
+                *[0] * 40,
             ],
         )
         self.assertEqual(collection_1.flags, 0)
         self.assertEqual(collection_1.min, 0.045)
         self.assertEqual(collection_1.max, 8)
 
+    def test_cumulative_aggregation_with_random_data(self) -> None:
+        histogram = _ExponentialBucketHistogramAggregation(Mock(), Mock())
+
+        def collect_and_validate() -> None:
+            result: ExponentialHistogramDataPoint = histogram.collect(
+                AggregationTemporality.CUMULATIVE, 0
+            )
+            buckets = result.positive.bucket_counts
+            scale = result.scale
+            index_start = result.positive.offset
+
+            for i in range(len(buckets)):
+                index = index_start + i
+                count = buckets[i]
+                lower_bound = 2 ** (index / (2**scale))
+                upper_bound = 2 ** ((index + 1) / (2**scale))
+                matches = 0
+                for value in values:
+                    if value > lower_bound and value <= upper_bound:
+                        matches += 1
+                assert (
+                    matches == count
+                ), f"index: {index}, count: {count}, scale: {scale}, lower_bound: {lower_bound}, upper_bound: {upper_bound}, matches: {matches}"
+
+            assert sum(buckets) + result.zero_count == len(values)
+            assert result.sum == sum(values)
+            assert result.count == len(values)
+            assert result.min == min(values)
+            assert result.max == max(values)
+            assert result.zero_count == len([v for v in values if v == 0])
+            assert scale >= 3
+
+        random = insecure_random.Random("opentelemetry2")
+        values = []
+        for i in range(2000):
+            value = random.randint(0, 1000)
```

Comment:
nit: should we try negative values too here to test the negative Bucket codepaths?

Reply:
I can do it in a followup, this pr is already big enough